type-generator/src/main/java/io/mesosphere/types/dtr/Main.java

```diff
@@ -0,0 +1,156 @@
+        options.addOption(o_project);
+        options.addOption(o_version);
+        options.addOption(o_repo);
+
+        // Parse options
+        try {
+            CommandLine cmd = parser.parse(options, args);
+
+            // Handle and bail early on hep argument
+            if (cmd.hasOption("help")) {
+                HelpFormatter formatter = new HelpFormatter();
+                formatter.printHelp( "dtr", options );
+                return;
+            }
+
+            // Create a repository that is going to be our access
+            // channel to the type registry.
+            Repository repo = Repository.fromURI(cmd.getOptionValue("repo"));
+
+            // Access the specified project & version
+            RepositoryView project = repo.openProjectTypes(
+                cmd.getOptionValue("project"),
+                cmd.getOptionValue("version")
+            );
+
+            // Load project
+            RAMLParser raml = new RAMLParser();
+            GlobalScope globalScope = new GlobalScope();
+            raml.loadProjectTypes(project, globalScope);
+
+            // Initialize template
+            RepositoryView compilerView = repo.openProjectCompiler(
+                cmd.getOptionValue("project"),
+                cmd.getOptionValue("version")
+            );
+            CompilerEngine engine = new StringTemplateEngine(compilerView);
+            CompiledFragment frag = engine.getCompiledFragment(CompiledFragmentType.FILE_CONTENTS);
+
+            // Display type registry
+            for (TypeScope scope: globalScope.childScopesOfType(FileScope.class)) {
+                for (Map.Entry<String, Type> t: scope.flatTypeMap().entrySet()) {
+                    Type type = t.getValue();
+                    StructuralType sType = type.getStructural();
+
+                    if (!(sType instanceof ScalarType) && !(sType instanceof ArrayType)) {
+                        String baseDir = "/Users/icharala/Develop/playground-scala/src/main/scala/types/";
+                        FileOutputStream fos = new FileOutputStream(baseDir + type.getId() + ".scala");
+                        PrintStream ps = new PrintStream(fos, false, "UTF-8");
+                        ps.println("package types");
+                        ps.println("");
+                        ps.println("import java.time.OffsetDateTime");
+                        ps.println("import play.api.libs.json._");
+                        ps.println();
+                        ps.flush();
+
+                        FileObjectTypeFragmentArguments t_args = new FileObjectTypeFragmentArguments(type, globalScope, engine);
+                        frag.write(fos, t_args);
+                        ps.println("");
+
+                        ps.close();
+                        fos.close();
+                    }
+                }
+            }
+
+        } catch (UnrecognizedOptionException e) {
+            System.err.println("ERROR: Unrecognized option " + e.getOption());
```

Comment:
Why not use a logger?

Reply:
True. Left over from the debugging times... 👍